pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -88,21 +88,21 @@ func (c *createConfig) createNewOSDsFromStatus(
 		return
 	}
 
-	for _, osd := range status.OSDs {
+	for i, osd := range status.OSDs {
 		if c.deployments.Exists(osd.ID) {
 			// This OSD will be handled by the updater
 			logger.Debugf("not creating deployment for OSD %d which already exists", osd.ID)
 			continue
 		}
 		if status.PvcBackedOSD {
 			logger.Infof("creating OSD %d on PVC %q", osd.ID, nodeOrPVCName)
-			err := createDaemonOnPVCFunc(c.cluster, osd, nodeOrPVCName, c.provisionConfig)
+			err := createDaemonOnPVCFunc(c.cluster, &status.OSDs[i], nodeOrPVCName, c.provisionConfig)
 			if err != nil {
 				errs.addError("%v", errors.Wrapf(err, "failed to create OSD %d on PVC %q", osd.ID, nodeOrPVCName))
 			}
 		} else {
 			logger.Infof("creating OSD %d on node %q", osd.ID, nodeOrPVCName)
-			err := createDaemonOnNodeFunc(c.cluster, osd, nodeOrPVCName, c.provisionConfig)
+			err := createDaemonOnNodeFunc(c.cluster, &status.OSDs[i], nodeOrPVCName, c.provisionConfig)
```

Comment:
why only i'th?

Reply:
`&status.OSDs[i]` is the same individual OSD that was being passed previously by the `osd` var. The change is needed so we can pass a pointer to the osd info. We can't just pass `&osd` because it's a loop variable.